metadata-ingestion/src/datahub/ingestion/source/vertexai.py

```diff
@@ -0,0 +1,497 @@
+                    yield from self._get_data_process_properties_workunit(model.training_job)
+
+                # create work unit for Model (= Model Version in VertexAI)
+                logger.info(f"Generating work unit for model (name: {model.display_name} id:{model.name})")
+                yield from self._get_ml_model_endpoint_workunit(model=model, model_version=model_version)
+
+
+    def _get_training_job_workunit(self) -> Iterable[MetadataWorkUnit]:
+        """
+        Fetches training jobs from Vertex AI and generates corresponding work units.
+        This method retrieves various types of training jobs from Vertex AI, including
+        CustomJob, CustomTrainingJob, CustomContainerTrainingJob, CustomPythonPackageTrainingJob,
+        AutoMLTabularTrainingJob, AutoMLTextTrainingJob, AutoMLImageTrainingJob, AutoMLVideoTrainingJob,
+        and AutoMLForecastingTrainingJob. For each job, it generates work units containing metadata
+        about the job, its inputs, and its outputs.
+        """
+        logger.info("Fetching a list of CustomJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.CustomJob.list())
+        logger.info("Fetching a list of CustomTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.CustomTrainingJob.list())
+        logger.info("Fetching a list of CustomContainerTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.CustomContainerTrainingJob.list())
+        logger.info("Fetching a list of CustomPythonPackageTrainingJob from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.CustomPythonPackageTrainingJob.list())
+        logger.info("Fetching a list of AutoMLTabularTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.AutoMLTabularTrainingJob.list())
+        logger.info("Fetching a list of AutoMLTextTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.AutoMLTextTrainingJob.list())
+        logger.info("Fetching a list of AutoMLImageTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.AutoMLImageTrainingJob.list())
+        logger.info("Fetching a list of AutoMLVideoTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.AutoMLVideoTrainingJob.list())
+        logger.info("Fetching a list of AutoMLForecastingTrainingJobs from VertexAI server")
+        yield from self._get_data_process_workunit(self.client.AutoMLForecastingTrainingJob.list())
+
+    def _get_data_process_workunit(self, jobs: List[_TrainingJob]) -> Iterable[MetadataWorkUnit]:
+        for job in jobs:
+            yield self._get_data_process_properties_workunit(job)
+            yield from self._get_job_output_workunit(job)
+            yield from self._get_job_input_workunit(job)
```

Comment:
generate the full data process entity in one go

maybe add `generate_training_job()` that calls each one individually